docs/json-codec-backend/codec-docs.md

```diff
@@ -0,0 +1,97 @@
+
+### How to use Generated Codecs
+Using an example , lets describe a model structure and look at how the generated codec for this particular model will be structured
+
+The model:
+
+	- reference-model
+      - src
+        - Morphir
+          - Reference
+	        - Model
+              - BooksAndRecords.elm  
+	- morphir.JSON
+	- morphir-ir.JSON
+
+
+Running the command  ```morphir-elm gen --include-codecs --copy-deps --output codecs/src``` to generate codecs means your codecs would be outputted into the ``` codecs/src```  folder in the structure:
+
+	- reference-model
+	  - codecs
+        - src
+          - morphir
+            - reference
+                - model
+                  - booksandrecords
+                    - Codec.scala   <----- This is your genereated codec 
+	- morphir.JSON
+	- morphir-ir.JSON
+
+The generated codecs ```Codec.scala```  has both encoders and decoders.
+
+#### Importing and Using generated Codecs
+
+In the example above the codecs where generated based on the ```BooksAndRecords.elm``` model and output in the codecs package. As an example lets say we want to use the generated codecs in a package called ```usecodecs``` with the following structure:
+
+```
+- usecodecs
+  - src
+     - UseCodecs.scala
+        
+```
+
+Inside ```UseCodecs.scala``` we need to import the codecs from the package in which it was generated ie: ```morphir.reference.model.booksandrecords``` and use the codecs from the ```Codec``` object. Below is an example of how to export and use a codec.
+
+```
+import morphir.reference.model.booksandrecords.Codec._
+
+object UseCodecs {
+    def main(args: Array[Strings]): Unit = {
+        val bookType = BookType("HardCover")
+        
+        println(encodeBookType(bookType) <--- using encoder
+        
+        println(decodeBookType(bookType) <--- using decoder
+    }
+}
+
+```
+It is required to import ```io.circe.Json``` in the situation where you want to use methods from the ```circe``` library. Libraries such as ```io.circe.parser``` and ```io.circe.generic``` are usually used to work with encoders and decoders, you would need to any additional library as a dependency to your project in order import them when using the codecs.
+
+
+
+### Some useful flags to use with ```morphir-elm make```
+
+The JSON codec backend generates codecs for only types , however an IR is made up of types and values and codecs would be generated for only types. Below are some optional flags you might want to use when running the ```morphir-elm make``` command
+
+1. `-t`,` --types-only` flag : Only include type information in the IR, no values.
+2. `-o`, `--output <path>` flag : Target file location where the Morphir IR will be saved.'
+3. `-f`, `--fallback-cli` : Make use of the old cli make function.
```

Comment:
I think we should explicitly spell out that `--types-only` should be used when you run `make` on the Morphir IR itself. This is not going to be used for any other model. The `-f` flag is also only used here because the new CLI doesn't support recursive types yet. These are unfortunate limitations that only apply to running `make` on the Morphir IR itself. When those issues are resolved we can remove these comments.